minetime: init at 1.4.12

dtzWill opened the pull request:
###### Motivation for this change

Builds on #54693, making small changes for adding more attributes.

Open to suggestions for better approach, but this works well for me :).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dtzWill committed:
appimage: plumb support for extra attributes to be specified

dtzWill committed:
minetime: init at 1.4.10, using appimageTools

tilpner commented on pkgs/build-support/appimage/default.nix:
```diff
@@ -46,17 +46,16 @@ rec {
       exec ./AppRun "$@"
     '';
-  });
+  } // (builtins.removeAttrs args [ "name" "src" "extraPkgs" ]));
```
Would `with builtins; attrNames (functionArgs wrapAppImage)` be preferable here?

dtzWill replied:
Oh, neat! I think so, no need to maintain two lists and perhaps is clearer re:intent.

dtzWill committed:
appimageTools: use functionArgs as suggested, use dynamicLinker attr

dtzWill commented:
* rebased onto master for final appimagetools that was merged
* switched to `functionArgs`, seems okay but take a look please
* I previously dropped `inherit name extraPkgs` in `wrapType1`  which I've restored, not sure what that's about.

re:third point maybe that isn't needed since `args` is merged in anyway, shrug.

dtzWill committed:
appimageTools: small simplification

dtzWill commented:
(added commit removing the inherit in both variants, FWIW)

dtzWill commented:
@GrahamcOfBorg eval

dtzWill commented:
Look okay re:appimagetools changes, @tilpner ? :)

dtzWill committed:
minetime: 1.4.10 -> 1.4.11

https://github.com/marcoancona/MineTime/releases/tag/v1.4.11

dtzWill committed:
minetime: 1.4.11 -> 1.4.12

carlosdagos commented:
Thanks for adding this one @dtzWill. Really looking forward to it getting merged :)

tilpner commented:
Oh, sorry, I forgot about this one. Yes, looks fine!

dtzWill commented:
It's up to 1.4.12 fwiw, I have that commit ready but can submit in a follow-up :).  But it's just a bump and hashfix happily.

dtzWill commented:
@GrahamcOfBorg eval

tilpner commented:
I had one remaining suspicion about at which levels `extraPkgs` is optional now. As it is now, this breaks #54696 with `called without required argument 'extraPkgs`.

dtzWill commented:
Locally merging the linked standardnotes PR results in a tree that evaluates `standardnotes` without that error, AFAICT.  I think maybe https://github.com/NixOS/nixpkgs/pull/56259/commits/58ce19e097a9bc099bf86b45ca9317fddb59ccd0 fixed that, if unintentionally :).  Can you confirm, or have any ideas why I can't reproduce? Thanks!

tilpner commented:
I'm not sure. I checked out standardnotes-appimagetools and picked all Appimage-related commits (`g cherry-pick 6dcdf27 056d827 58ce19e`).

I think 58ce19e097a9bc099bf86b45ca9317fddb59ccd0 is the one breaking it, not fixing it.

dtzWill committed:
minetime: 1.4.12 -> 1.5.1, better entrypoint name, drop unused 32bit

https://github.com/marcoancona/MineTime/releases/tag/v1.5.1
https://github.com/marcoancona/MineTime/releases/tag/v1.5.0

dtzWill commented:
@GrahamcOfBorg build minetime standardnotes joplin-desktop

dtzWill commented:
Well regardless rebase + some new changes and all is well re:eval.

\o/

Look good, @tilpner ? :)

dtzWill commented:
> I think [58ce19e](https://github.com/NixOS/nixpkgs/commit/58ce19e097a9bc099bf86b45ca9317fddb59ccd0) is the one breaking it, not fixing it.

Sorry about this, you're right.  See comments on #54696; will probably PR if don't see that go through soon.
